convert2rhel/utils.py

```diff
@@ -717,14 +717,45 @@ def download_pkg(
     if ret_code != 0:
         loggerinst.warning("Output from the yumdownloader call:\n%s" % (output))
 
-        if not "CONVERT2RHEL_UNSUPPORTED_INCOMPLETE_ROLLBACK" in os.environ:
+        # Note: Checking toolopts here is a temporary solution. We need to
+        # restructure this to raise an exception on error and have the caller
+        # handle whether to use INCOMPLETE_ROLLBACK to do something for several
+        # reasons:
+        # (1) utils should be simple functions that take input and produce
+        #     output from it. Having knowledge of things specific to the
+        #     program (for instance, the environment variable that convert2rhel
+        #     uses) makes the utils depend on the specific place that they are
+        #     run instead.
+        # (2) Where an error condition arises, they should "return" that to the
+        #     caller to decide how to handle it by using an exception. Handling
+        #     it inside the function ties us to one specific behaviour on
+        #     error. (For instance, the incomplete rollback message here ties
+        #     downloading packages and performing rollbacks. But what about
+        #     downloading packages that are not tied to rollbacks. Maybe we
+        #     have to download a package in order for insights or
+        #     subscription-manager to run. In those cases, we either cannot use
+        #     this function or we might show the user a misleading message).
+        # (3) Functions in utils should be free of other dependencies within
+        #     convert2rhel.  That allows us to use utils with no fear of
+        #     circular dependency issues.
+        # (4) Making the choices here mean that when used inside of the Action
+        #     framework, we are limited to returning a FAILURE for the Action
+        #     plugin whereas returning SKIP would be more accurate.
+        from convert2rhel import toolopts
+
+        if (
+            "CONVERT2RHEL_UNSUPPORTED_INCOMPLETE_ROLLBACK" not in os.environ
+            or toolopts.tool_opts.activity != "conversion"
+        ):
             loggerinst.critical(
                 "Couldn't download the %s package. This means we will not be able to do a"
                 " complete rollback and may put the system in a broken state.\n"
                 "Check to make sure that the %s repositories are enabled"
                 " and the package is updated to its latest version.\n"
                 "If you would rather ignore this check set the environment variable"
-                " 'CONVERT2RHEL_UNSUPPORTED_INCOMPLETE_ROLLBACK'." % (pkg, system_info.name)
+                " 'CONVERT2RHEL_UNSUPPORTED_INCOMPLETE_ROLLBACK'. Note that the"
+                " environment variable is only consulted when actually converting a"
+                " system. It is disabled when performing a pre-conversion analysis." % (pkg, system_info.name)
```

Comment:
This is redundant I feel. I do not think we need to specify this as the analyze command is separate workflow in a way

Reply:
Agreed with @SpyTec, it seems a bit redundant in this case.

Reply:
Currently this error message is output in both the analyze and convert workflow.  So without this portion of the message, the user would run `CONVERT2RHEL_EXPERIMENTAL_ANALYSIS=1 convert2rhel` , get the error message, rerun it as `CONVERT2RHEL_UNSUPPORTED_INCOMPLETE_ROLLBACK=1 CONVERT2RHEL_EXPERIMENTAL_ANALYSIS=1 convert2rhel` and then be left wondering why they got the same error message telling them to use `CONVERT2RHEL_UNSUPPORTED_INCOMPLETE_ROLLBACK=1`.

We could probably give a different error message when running under analyze than from convert... We probably do need to mention that `INCOMPLETE_ROLLBACK` isn't supported in the analyze case but we could forgo adding this in the convert case.

Reply:
This has been updated so that there is a separate error message for analysis and conversions

Reply:
LGTM